src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -349,8 +349,9 @@ export class CellModel extends Disposable implements ICellModel {
 	private attachImageFromSource(newSource: string | string[]): string | string[] {
 		if (!Array.isArray(newSource) && this.isValidBase64OctetStream(newSource)) {
 			let results;
-			while ((results = validBase64OctetStreamRegex.exec(newSource)) !== null) {
-				let imageName = this.addAttachment(results[1], results[0], 'image.png');
+			// only replace the base64 value if it's from markdown [](base64value) not html tags <img src="base64value"
+			while ((results = validBase64OctetStreamRegex.exec(newSource)) !== null && newSource.substring(newSource.indexOf(results[0]) - 9, newSource.indexOf(results[0]) - 2) !== 'img src') {
```

Comment:
Instead of checking for this logic here, can we consolidate logic in notebookMarkdown.ts? https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/notebook/browser/outputs/notebookMarkdown.ts#L74

Haven't looked into this myself, just took a quick look here. Ideally, the cell model knows about attachments, but doesn't have to actually do any html parsing.

Reply:
I tried that but it doesn't hit that flow at all when an image is pasted. Going to debug and check where it's building the html part.